pox/core.py

```diff
@@ -379,18 +379,25 @@ def _add_signal_handlers (self):
     if not isinstance(threading.current_thread(), threading._MainThread):
       raise RuntimeError("add_signal_handers must be called from MainThread")
 
+    self._install_signal_handler(signal.SIGTERM, self._signal_handler_SIGTERM)
     try:
-      previous = signal.getsignal(signal.SIGHUP)
-      signal.signal(signal.SIGHUP, self._signal_handler_SIGHUP)
-      if previous != signal.SIG_DFL:
-        log.warn('Redefined signal handler for SIGHUP')
+      self._install_signal_handler(signal.SIGHUP, self._signal_handler_SIGHUP)
     except (AttributeError, ValueError):
       # SIGHUP is not supported on some systems (e.g., Windows)
       log.debug("Didn't install handler for SIGHUP")
 
+  def _install_signal_handler(self, signum, handler):
+    previous = signal.getsignal(signum)
+    signal.signal(signum, handler)
+    if previous != signal.SIG_DFL:
+      log.warn('Redefined signal handler for ' + str(signum))
```

Comment:
Maybe try to get a nice signal name using something like...
```
{x.value:x.name for x in signal.Signals}.get(signum, f"#{signum}")
```
?

(Or something more obvious and more verbose.)

Reply:
I ended up using just `signum.name`.

The rationale is `signal.signal()` would fail with OSError if unknown signal number passed. So I assume type of `signum` is enum `signal.Signals`. Currently we call this method with such arguments only.

Does it work?